avidemux_plugins/ADM_videoFilters6/asharp/qt4/asharp.ui

```diff
@@ -125,22 +113,70 @@
      </property>
     </widget>
    </item>
-   <item row="6" column="0" colspan="4">
+   <item row="6" column="0" colspan="6">
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
-   <item row="0" column="2">
-    <widget class="QDoubleSpinBox" name="doubleSpinBoxTreshold"/>
+   <item row="0" column="3">
+    <widget class="QDoubleSpinBox" name="doubleSpinBoxTreshold">
+     <property name="maximum">
+      <double>32.000000000000000</double>
+     </property>
+    </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
-      <string>Strength</string>
+      <string>Adaptive strength</string>
      </property>
     </widget>
    </item>
-   <item row="4" column="0">
+   <item row="4" column="0" colspan="6">
     <layout class="QHBoxLayout" name="toolboxLayout"/>
    </item>
+   <item row="0" column="1">
+    <widget class="QSlider" name="horizontalSliderTreshold">
+     <property name="maximum">
+      <number>3200</number>
+     </property>
+     <property name="orientation">
+      <enum>Qt::Horizontal</enum>
+     </property>
+    </widget>
+   </item>
+   <item row="2" column="0">
+    <widget class="QLabel" name="label_3">
+     <property name="text">
+      <string>Block adaptive</string>
+     </property>
+    </widget>
+   </item>
+   <item row="2" column="1">
+    <widget class="QSlider" name="horizontalSliderBlock">
+     <property name="maximum">
+      <number>400</number>
+     </property>
+     <property name="orientation">
+      <enum>Qt::Horizontal</enum>
+     </property>
+    </widget>
+   </item>
+   <item row="1" column="1">
+    <widget class="QSlider" name="horizontalSliderStrength">
+     <property name="maximum">
+      <number>1600</number>
+     </property>
+     <property name="orientation">
+      <enum>Qt::Horizontal</enum>
+     </property>
+    </widget>
+   </item>
+   <item row="1" column="5">
+    <widget class="QLabel" name="label_4">
+     <property name="text">
+      <string>(0 == disabled)</string>
```

Comment:
Could we please do without this label (and for optical reasons probably also without the entire fifth column)? I feel it states the obvious. If you feel some hint to be necessary, maybe using setSpecialValueText() to confirm for the user that "0" means "disabled" instead?

Reply:
OK, a will remove it.

Reply:
Thanks!

Reply:
If Strength is set to 0, the effect is the same as when it is set to maximum. To disable the filter, it is actually necessary to set it to a small vaue > 0. Do you have an idea what is wrong with the backend code?

`void asharp_run_c(uc* planeptr, int pitch, int height, int width, int T, int D, int B, int B2, bool bf, uint8_t *lineptr);`

indeed gets D set to 0 if param.d is equal zero and 8192 when set to 16 (maximum).

Reply:
According to the avisynth documentation this is the expected behaviour. If > 0 then the unsharp mask applied adaptively, if disabled (== 0) the mask get applied as is.
It would may be the proper way, to add a checkbox, to enable or disable this numeric entry.
Reading this documentation, i came to the conclusion that an another checkbox should be applied on the Block adaptive entry.